packages/react-ai/src/components/AIConversation/views/Controls/FieldControl.tsx

```diff
@@ -0,0 +1,256 @@
+import React from 'react';
+
+import { withBaseElementProps } from '@aws-amplify/ui-react-core/elements';
+import { ConversationInputContext } from '../../context';
+import { AIConversationElements } from '../../context/elements';
+import { AttachFileControl } from './AttachFileControl';
+import { MessagesContext } from '../../context';
+import { AttachmentListControl } from './AttachmentListControl';
+import { SendMessageContext } from '../../context/SendMessageContext';
+import { ConversationMessageContent, InputContent } from '../../../../types';
+import {
+  convertResponseComponentsToToolConfiguration,
+  ResponseComponentsContext,
+} from '../../context/ResponseComponentsContext';
+import { ControlsContext } from '../../context/ControlsContext';
+import { getImageTypeFromMimeType } from '../../utils';
+import { LoadingContext } from '../../context/LoadingContext';
+
+const {
+  Button,
+  Icon,
+  Label: LabelElement,
+  TextArea,
+  View,
+} = AIConversationElements;
+
+const FIELD_BLOCK = 'ai-field';
+
+const SendIcon = withBaseElementProps(Icon, {
+  className: `${FIELD_BLOCK}__icon`,
+  variant: 'send-message',
+});
+
+const SendButtonBase = withBaseElementProps(Button, {
+  'aria-label': 'Send message',
+  className: `${FIELD_BLOCK}__button ${FIELD_BLOCK}__button--send`,
+});
+
+const SendButton: typeof SendButtonBase = React.forwardRef(
+  function SendButton(props, ref) {
+    const { input } = React.useContext(ConversationInputContext);
+    const isLoading = React.useContext(LoadingContext);
+    const hasInput = !!input?.text || !!input?.files?.length;
+
+    return (
+      <SendButtonBase
+        {...props}
+        // we intentionally || in the case where isLoading is false we should use the value of hasInput
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        disabled={isLoading || !hasInput}
```

Comment:
Think we can default isLoading to false where we destructure it?

Reply:
In cases where `isLoading` is undefined we should use the value of `!hasInput` though, we shouldn't short circuit to the value false from `isLoading`

Reply:
`||` won't short circuit on `false` though right?

Reply:
Ah i meant `true`, in the false case we just defer to `!hasInput` because if it's not loading we don't need to disable the input

Reply:
Right, so if we default `isLoading` to `false` the linting rule will stop yelling since `isLoading` will no longer resolve to potentially `undefined` without impacting the second condition of `hasInput`